### ISSUE TITLE
[Bug]: Tooltip not visible for newly added items in <Dropdown>

### ISSUE THREAD

**Issue by joshualiu** [carbon-design-system/carbon#11762]
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.6.0

### React version

v17

### Description

When adding new items to the `<Dropdown>` component `items` list, the newly added items don't show a tooltip.

<img width="412" alt="Screen Shot 2022-07-07 at 10 49 00 AM" src="https://user-images.githubusercontent.com/25513105/177804966-460b39ed-62f8-438b-ba8f-4f27d955a335.png">

<img width="365" alt="Screen Shot 2022-07-07 at 10 49 21 AM" src="https://user-images.githubusercontent.com/25513105/177804948-31d1aafd-a8a6-4526-9537-3ca500ce527b.png">

Note: this can be reproduced in `carbon-components-react` `7.57.0` as well

### Reproduction/example

https://codesandbox.io/s/floral-violet-wl7idz

### Steps to reproduce

- Go to CodeSandbox: https://codesandbox.io/s/floral-violet-wl7idz
- Open the default dropdown, the default option has a tooltip on hover
- Click the button above to add more options to the `items` list
- The tooltip is not visible for newly added items, only the default one.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
Hey @joshualiu, can confirm I'm able to reproduce this issue and we'll take a look at it in an upcoming sprint.

Possibly tied to the `title` being initialized in the `isTruncated` effect

https://github.com/carbon-design-system/carbon/blob/fd388d1c7804d1806ac64f510afb0f63e8f39800/packages/react/src/components/ListBox/ListBoxMenuItem.js#L45
